File: src/test/java/ml/glassmc/test/shard/TestReference.java
package ml.glassmc.test.shard;

import ml.glassmc.loader.Hook;
import ml.glassmc.loader.Reference;
import ml.glassmc.loader.hook.ClientHookType;

import java.util.HashMap;
import java.util.Map;

public class TestReference implements Reference {

    @Override
    public Map<Class<?>, Class<? extends Hook>> getHooks() {
        return new HashMap<Class<?>, Class<? extends Hook>>() {
            {
                put(ClientHookType.class, TestMainHook.class);
            }
        };
    }

}

File: impl/client/src/test/kotlin/ml/glassmc/loader/client/test/shard/TestReference.kt
package ml.glassmc.loader.client.test.shard

import ml.glassmc.loader.Listener
import ml.glassmc.loader.Reference
import ml.glassmc.loader.client.hook.ClientInitializeHook
import java.util.HashMap

class TestReference : Reference {

    override val listeners: Map<Class<*>, Class<out Listener>>
        get() = object: HashMap<Class<*>, Class<out Listener>>() {
            init {
                put(ClientInitializeHook::class.java, TestClientInitializeListener::class.java)
            }
        }

}